tests/benchmark/bm_utils.h

```diff
@@ -1,55 +1,192 @@
+#pragma once
+
 #include <benchmark/benchmark.h>
 #include <random>
 #include <unistd.h>
+#include <istream>
 #include "VecSim/vec_sim.h"
 #include "VecSim/vec_sim_interface.h"
 #include "VecSim/query_results.h"
 #include "VecSim/utils/arr_cpp.h"
 #include "VecSim/algorithms/brute_force/brute_force.h"
 #include "VecSim/algorithms/hnsw/hnsw_single.h"
+#include "bm_common.h"
 
-class BM_VecSimBasics : public benchmark::Fixture {
+class BM_VecSimGeneral : public benchmark::Fixture {
```

Comment:
this class includes all data and members functions that are type independent

Reply:
rename the file please